neo4j/io/_bolt3.py

```diff
@@ -225,11 +226,8 @@ def run(self, query, parameters=None, mode=None, bookmarks=None,
                 extra["tx_metadata"] = dict(metadata)
             except TypeError:
                 raise TypeError("Metadata must be coercible to a dict")
-        if timeout:
-            try:
-                extra["tx_timeout"] = int(1000 * timeout)
-            except TypeError:
-                raise TypeError("Timeout must be specified as a number of seconds")
+        if timeout or timeout == 0:
```

Comment:
Not sure if I'm reading this correctly. Is timeout a float here and being used in an equality check?

Reply:
I applied a small code change, but it can still be a comparison of a float (depending on what type the user passes to the driver). However, to my knowledge, only the only float that's falsy is `float(0)`. So the first half half of the condition would short circuit for any other value.